### ISSUE TITLE
New Feature: OpenAPICustomiser

### ISSUE THREAD

**Issue by davidmelia** [springdoc/springdoc-openapi#92]
Hi,

I have a requirement to add a common header to every API call so I've come up with the following: 

1) Turn off the default springdoc OpenApiResource:
springdoc.api-docs.enabled=false
2) Create a customisable interface
```
public interface OpenAPICustomiser {
  public void customise(OpenAPI openApi);
}
```
3) Extend OpenApiResource and add the optional customiser check.
```
@RestController
public class CustomisableOpenApiResource extends org.springdoc.api.OpenApiResource {

private final Optional<List<OpenAPICustomiser>> openApiCustomisers;
. . .
  @Override
  protected OpenAPI getOpenApi() {
    final OpenAPI openApi = super.getOpenApi();
    if (openApiCustomisers.isPresent()) {
      openApiCustomisers.get().stream().forEach(openApiCustomiser -> openApiCustomiser.customise(openApi));
    }
    return openApi;
  }

```

4) Add one to many customisations - following shows adding a common header to every operation:
```
  @Bean
  public OpenAPICustomiser consumerTypeHeaderOpenAPICustomiser() {
    return openApi -> openApi.getPaths().values().stream().flatMap(pathItem -> pathItem.readOperations().stream())
        .forEach(operation -> operation.addParametersItem(new HeaderParameter().$ref("#/components/parameters/myConsumerTypeHeader")));
  }
```

Would it be possible to provide this ability in the default org.springdoc.api.OpenApiResource? 

Thanks

**Comment by springdoc** [comment]
Hi, 

 Can you propose a pull request, with this feature ?

**Comment by davidmelia** [comment]
sure will have a go this week.  Thanks

**Comment by davidmelia** [comment]
@springdoc have a look at 

https://github.com/springdoc/springdoc-openapi/pull/93

Thanks

**Comment by springdoc** [comment]
Hi, 

 Thank you for your contribution.
 This feature, will be available on v1.1.43 of springdoc-openapi.
